packages/d/discord/xmake.lua

```diff
@@ -4,22 +4,43 @@ package("discord")
     set_description("Whether you’re part of a school club, gaming group, worldwide art community, or just a handful of friends that want to spend time together, Discord makes it easy to talk every day and hang out more often.")
 
     add_versions("2.5.6", "426eb5fa70647d884f461c63825b63668349efb4bc68a16e70bc4a24e119b92e")
+    add_versions("3.2.1", "6757bb4a1f5b42aa7b6707cbf2158420278760ac5d80d40ca708bb01d20ae6b4")
+
+    add_patches("2.5.6", path.join(os.scriptdir(), "patches", "2.5.6", "add_include_cstdint_to_typeh.patch"), "d67e7a0e6aa4a90a7690b3a2f86342becbb052fcd1a38395151cdcb29804ca83")
+    add_patches("3.2.1", path.join(os.scriptdir(), "patches", "3.2.1", "add_include_cstdint_to_typeh.patch"), "31f9422879ae299600607d10606efd807a8acb32697f0c9d9bbdd2e3969935c4")
 
     set_urls("https://dl-game-sdk.discordapp.net/$(version)/discord_game_sdk.zip")
 
-    on_install("windows", function (package)
+    on_install("windows", "linux", "macosx", function (package)
         os.cp("cpp/*.h", package:installdir("include"))
         os.cp("c/*.h", package:installdir("include"))
-        os.cp("Release/*.lib", package:installdir("lib"))
-        if package:is_arch("x64") then
-            os.cp("lib/x86_64/discord_game_sdk.dll", package:installdir("bin"))
-            os.cp("lib/x86_64/discord_game_sdk.dll.lib", package:installdir("lib"))
-        else
-            os.cp("lib/x86/discord_game_sdk.dll", package:installdir("bin"))
-            os.cp("lib/x86/discord_game_sdk.dll.lib", package:installdir("lib"))
-        end
 
         local configs = {}
+        if is_plat("windows") then
+            if package:is_arch("x64") then
+                os.cp("lib/x86_64/discord_game_sdk.dll", package:installdir("bin"))
+                os.cp("lib/x86_64/discord_game_sdk.dll.lib", package:installdir("lib"))
+                configs.linkflags = "-llib/x86_64/discord_game_sdk.dll"
```

Comment:
no linkflags, `configs.linkdirs = "lib/x86_64"` and `configs.links = "discord_game_sdk.dll"`